realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -0,0 +1,286 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.realm.entities.AllTypes;
+import io.realm.rule.RunInLooperThread;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+import static junit.framework.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class RealmIntegerTests {
+
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    @Rule
+    public final RunInLooperThread looperThread = new RunInLooperThread();
+
+    private Realm realm;
+
+    @Before
+    public void setUp() throws Exception {
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
+        realm = Realm.getInstance(realmConfig);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (realm != null) {
+            if (realm.isInTransaction()) {
+                realm.cancelTransaction();
+            }
+            realm.close();
+        }
+    }
+
+    /**
+     * Validate basic constructor parameters and functions set, increment and decrement.
+     */
+    @Test
+    public void basic() {
+        RealmInteger ri1 = new RealmInteger(10);
+        RealmInteger ri2 = new RealmInteger("10");
+        assertEquals(ri1, ri2);
+
+        ri1.set(15);
+        ri1.decrement(2);
+        ri2.increment(3);
+        assertEquals(ri1, ri2);
+    }
+
+    /**
+     * Validate various getters.
+     * Expected behaviour is that gets of smaller sized quantities should truncate on the left:
+     * that they should return exactly the rightmost N bits of the underlying value.
+     * <p>
+     * Caution.  The assertion functions will cast back up to int
+     * if either arg is an int.  That will cause sign extension.
+     */
+    @Test
+    public void getters() {
+        RealmInteger ri = new RealmInteger(0x5555444433332211L);
+
+        // positive
+        assertEquals(0x5555444433332211L, ri.longValue());
+        assertEquals(0x033332211, ri.intValue());
+        assertEquals((short) 0x02211, ri.shortValue());
+        assertEquals((byte) 0x011, ri.byteValue());
+
+        assertEquals(6.1488962E18F, ri.floatValue());
+        assertEquals(6.1488959259517348E18, ri.doubleValue());
+
+        // negative
+        ri.set(0x8888444483338281L);
+        assertEquals(0x8888444483338281L, ri.longValue());
+        assertEquals(0x083338281, ri.intValue());
+        assertEquals((short) 0xf8281, ri.shortValue());
+        assertEquals((byte) 0xf81, ri.byteValue());
```

Comment:
In current code, we need to think about the meaning of each different values.
I think that casting the same value is much easier to understand.